src/System.Net.Sockets.Legacy/tests/FunctionalTests/DisposedSocketTests.cs

```diff
@@ -0,0 +1,229 @@
+        {
+            SocketError errorCode;
+            Assert.Throws<ObjectDisposedException>(() => GetDisposedSocket().EndReceive(null, out errorCode));
+        }
+
+        [Fact]
+        public void BeginReceiveFrom_Throws_ObjectDisposedException()
+        {
+            EndPoint remote = new IPEndPoint(IPAddress.Loopback, 1);
+            Assert.Throws<ObjectDisposedException>(() => GetDisposedSocket().BeginReceiveFrom(s_buffer, 0, s_buffer.Length, SocketFlags.None, ref remote, TheAsyncCallback, null));
+        }
+
+        [Fact]
+        public void EndReceiveFrom_Throws_ObjectDisposed()
+        {
+            EndPoint remote = new IPEndPoint(IPAddress.Loopback, 1);
+            Assert.Throws<ObjectDisposedException>(() => GetDisposedSocket().EndReceiveFrom(null, ref remote));
+        }
+
+        [Fact]
+        public void BeginReceiveMessageFrom_Throws_ObjectDisposedException()
+        {
+            EndPoint remote = new IPEndPoint(IPAddress.Loopback, 1);
+            Assert.Throws<ObjectDisposedException>(() => GetDisposedSocket().BeginReceiveMessageFrom(s_buffer, 0, s_buffer.Length, SocketFlags.None, ref remote, TheAsyncCallback, null));
+        }
+
+        [Fact]
+        public void EndReceiveMessageFrom_Throws_ObjectDisposed()
+        {
+            SocketFlags flags = SocketFlags.None;
+            EndPoint remote = new IPEndPoint(IPAddress.Loopback, 1);
+            IPPacketInformation packetInfo;
+            Assert.Throws<ObjectDisposedException>(() => GetDisposedSocket().EndReceiveMessageFrom(null, ref flags, ref remote, out packetInfo));
+        }
+
+        [Fact]
+        public void BeginAccept_Throws_ObjectDisposed()
+        {
+            Assert.Throws<ObjectDisposedException>(() => GetDisposedSocket().BeginAccept(TheAsyncCallback, null));
+        }
+
+        [Fact]
+        public void BeginAccept_Int_Throws_ObjectDisposed()
+        {
+            Assert.Throws<ObjectDisposedException>(() => GetDisposedSocket().BeginAccept(0, TheAsyncCallback, null));
+        }
+
+        [Fact]
+        public void BeginAccept_Socket_Int_Throws_ObjectDisposed()
+        {
+            Assert.Throws<ObjectDisposedException>(() => GetDisposedSocket().BeginAccept(null, 0, TheAsyncCallback, null));
+        }
+
+        [Fact]
+        public void EndAccept_Throws_ObjectDisposed()
+        {
+            Assert.Throws<ObjectDisposedException>(() => GetDisposedSocket().EndAccept(null));
+        }
+
+        [Fact]
+        public void EndAccept_Buffer_Throws_ObjectDisposed()
+        {
+            byte[] buffer;
+            Assert.Throws<ObjectDisposedException>(() => GetDisposedSocket().EndAccept(out buffer, null));
+        }
+
+        [Fact]
+        public void EndAccept_Buffer_Int_Throws_ObjectDisposed()
+        {
+            byte[] buffer;
+            int received;
+            Assert.Throws<ObjectDisposedException>(() => GetDisposedSocket().EndAccept(out buffer, out received, null));
+        }
+    }
+}
```

Comment:
I don't think we need or should spend time adding a full set of such tests, but consider adding a few spot-check tests to verify that the right exception gets thrown when multiple things are wrong, e.g. when an invalid argument is passed to a method on a disposed socket.  This would help to catch refactorings in the future that change the order of such checks in the implementation.